app/utils/chart-config/versioning.ts

```diff
@@ -1801,6 +1825,26 @@ export const configuratorStateMigrations: Migration[] = [
 
       newConfig.chartConfigs = chartConfigs;
 
+      const dataFilters = newConfig.dashboardFilters.dataFilters;
+
+      for (const cId of dataFilters.componentIds) {
+        dataFilters.componentIds = [
+          ...dataFilters.componentIds.filter((c: string) => c !== cId),
+          splitComponentId(cId).unversionedComponentIri,
+        ];
+      }
+
+      for (const fId of Object.keys(dataFilters.filters)) {
+        dataFilters.filters[
+          splitComponentId(fId).unversionedComponentIri as string
+        ] = dataFilters.filters[fId];
+        delete dataFilters.filters[fId];
+      }
+
+      dataFilters.componentIris = dataFilters.componentIds;
+      delete dataFilters.componentIds;
+      newConfig.dashboardFilters.dataFilters = dataFilters;
+
```

Comment:
Yeah, I don't like this either, but I think abstracting this would just move this (or very similar) logic to a different place in the codebase? Maybe there could be a visitor pattern that would go through every possible field and return a `[key, value]` pair in case the property contains "componentIri"?

But as this is generally one-off thing and I feel like abstracting this would be a bit complex, maybe it's fine to leave it as it is? 🤔